src/domain/Processors/VehicleProcessor.ts

```diff
@@ -64,16 +64,22 @@ export abstract class VehicleProcessor<T extends Vehicle> {
   protected updateVehicleIdentifiers(existingVehicle: T, updatedVehicle: T): T {
     const { primaryVrm } = updatedVehicle;
     const previousPrimaryVrm = existingVehicle.primaryVrm;
-    updatedVehicle.secondaryVrms = existingVehicle.secondaryVrms;
+    updatedVehicle.secondaryVrms = existingVehicle.secondaryVrms ? Object.assign([], existingVehicle.secondaryVrms) : undefined
```

Comment:
Because you are not merging 2 object and instead assigning a new copy of `existingVehicle.secondaryVrms`, I would recommend using `cloneDeep()`.
```suggestion
    updatedVehicle.secondaryVrms = existingVehicle.secondaryVrms ? cloneDeep(existingVehicle.secondaryVrms) : undefined
```